### PR TITLE
SPARKS-162 Add keyspce level/cluster level Conf settings

### PR DESCRIPTION
**Pull request by alexliu68**
Store keyspace level settings as cluster.keyspace/xx.xx.xx.xx
and cluseter level settings as cluster/xx.xx.xx.xx in
SQLConf

Retrieve settings in the order of
table level -> keyspace -> cluster -> default. Use the one
first available.

### CODE REVIEW

File: spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraSourceRelation.scala [hunk 1]
```diff
@@ -168,4 +164,28 @@ object CassandraSourceRelation {
...
+  /**
+   * Consolidate Cassandra conf settings in the order of table level -> keyspace level ->
+   * cluster level -> default. Use the first available setting
+   */
+  def consolidateConfs(sqlContext: SQLContext, tableRef: TableRef, cassandraConfs: Map[String, String]) : SparkConf = {
+    //Default settings
+    val conf = sqlContext.sparkContext.getConf.clone()
+    //Keyspace/Cluster level settings
+    val sqlConf = sqlContext.getAllConfs
+    for (prop <- DefaultSource.confProperties) {
```
Comment (RussellSpitzer) [inline spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraSourceRelation.scala]: tiny change, I would have prop be called defaultProp so it is more clear in the following if statements

**Comment by RussellSpitzer** [comment]
+1

**Comment by alexliu68** [comment]
merge this into SPARKC-112

**Comment by pkolaczk** [comment]
This commit appears to try launching SparkContext in the unit test suite, which starts services and binds to ports. This is not acceptable in the unit test suite. Moreover this attempt fails, despite the test passing successfully.  

Is it really needed for such a simple feature? Why is the config tightly coupled to SparkSQLContext?  

Therefore I reverted the merge. Please fix it and submit for review again.
Also, for the future, please don't merge if you didn't get +1 from all the reviewers mentioned on the ticket.

```
[info] ConsolidateSettingsSpec:
Starting SparkContext with the following configuration:
spark.app.name=Test
spark.cassandra.connection.host=127.0.0.1
spark.cleaner.ttl=3600
spark.master=local[*]
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
15/05/21 09:39:47 INFO SparkContext: Running Spark version 1.3.1
15/05/21 09:39:47 WARN Utils: Your hostname, m4600 resolves to a loopback address: 127.0.1.1; using 10.0.0.103 instead (on interface eth0)
15/05/21 09:39:47 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
15/05/21 09:39:47 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
15/05/21 09:39:47 INFO SecurityManag
er: Changing view acls to: pkolaczk
15/05/21 09:39:47 INFO SecurityManager: Changing modify acls to: pkolaczk
15/05/21 09:39:47 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: Set(pkolaczk); users with modify permissions: Set(pkolaczk)
15/05/21 09:39:48 INFO Slf4jLogger: Slf4jLogger started
15/05/21 09:39:48 INFO Remoting: Starting remoting
15/05/21 09:39:48 INFO Remoting: Remoting started; listening on addresses :[akka.tcp://sparkDriver@10.0.0.103:52680]
15/05/21 09:39:48 INFO Utils: Successfully started service 'sparkDriver' on port 52680.
15/05/21 09:39:48 INFO SparkEnv: Registering MapOutputTracker
15/05/21 09:39:48 INFO SparkEnv: Registering BlockManagerMaster
15/05/21 09:39:48 INFO DiskBlockManager: Created local directory at /tmp/spark-6ef84e62-d13c-4843-82e9-561a6e90ae4a/blockmgr-f7de7e92-2219-4cf5-98e2-fe87a5f8cf1a
15/05/21 09:39:48 INFO MemoryStore: MemoryStore started with capacity 2.8 GB
15/05/21 09:39:48 INFO HttpFileServer: HTTP File server directory is /tmp/spark-e57d4780-9eb2-4dde-88e1-4975008072a5/httpd-8a14f670-652d-4bf1-8671-6cfe5e18e6dc
15/05/21 09:39:48 INFO HttpServer: Starting HTTP Server
15/05/21 09:39:48 INFO Server: jetty-8.y.z-SNAPSHOT
15/05/21 09:39:48 INFO AbstractConnector: Started SocketConnector@0.0.0.0:54029
15/05/21 09:39:48 INFO Utils: Successfully started service 'HTTP file server' on port 54029.
15/05/21 09:39:48 INFO SparkEnv: Registering OutputCommitCoordinator
15/05/21 09:39:48 INFO Server: jetty-8.y.z-SNAPSHOT
15/05/21 09:39:48 INFO AbstractConnector: Started SelectChannelConnector@0.0.0.0:4040
15/05/21 09:39:48 INFO Utils: Successfully started service 'SparkUI' on port 4040.
15/05/21 09:39:48 INFO SparkUI: Started SparkUI at http://10.0.0.103:4040
15/05/21 09:39:48 INFO Executor: Starting executor ID <driver> on host localhost
15/05/21 09:39:48 INFO AkkaUtils: Connecting to HeartbeatReceiver: akka.tcp://sparkDriver@10.0.0.103:52680/user/HeartbeatReceiver
15/05/21 09:39:49 INFO NettyBlockTransferService: Server created on 43196
15/05/21 09:39:49 INFO BlockManagerMaster: Trying to register BlockManager
15/05/21 09:39:49 INFO BlockManagerMasterActor: Registering block manager localhost:43196 with 2.8 GB RAM, BlockManagerId(<driver>, localhost, 43196)
15/05/21 09:39:49 INFO BlockManagerMaster: Registered BlockManager
15/05/21 09:39:49 INFO SparkContext: Running Spark version 1.3.1
15/05/21 09:39:49 INFO SecurityManager: Changing view acls to: pkolaczk
15/05/21 09:39:49 INFO SecurityManager: Changing modify acls to: pkolaczk
15/05/21 09:39:49 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: Set(pkolaczk); users with modify permissions: Set(pkolaczk)
15/05/21 09:39:49 INFO Slf4jLogger: Slf4jLogger started
15/05/21 09:39:49 INFO Remoting: Starting remoting
15/05/21 09:39:49 INFO Remoting: Remoting started; listening on addresses :[akka.tcp://sparkDriver@localhost:51701]
15/05/21 09:39:49 INFO Utils: Successfully started service 'sparkDriver' on port 51701.
15/05/21 09:39:49 INFO SparkEnv: Registering MapOutputTracker
15/05/21 09:39:49 INFO SparkEnv: Registering BlockManagerMaster
15/05/21 09:39:49 INFO DiskBlockManager: Created local directory at /tmp/spark-ea9fcb10-064e-45da-8ecc-862935079768/blockmgr-ca6af5a9-a187-45a1-a424-7871784825c3
15/05/21 09:39:49 INFO MemoryStore: MemoryStore started with capacity 2.8 GB
15/05/21 09:39:49 INFO HttpFileServer: HTTP File server directory is /tmp/spark-73d3dc78-936b-4e0b-b520-c55a2acd07ab/httpd-c79e1bab-a7e7-4004-87a5-ceeb649adadd
15/05/21 09:39:49 INFO HttpServer: Starting HTTP Server
15/05/21 09:39:49 INFO Server: jetty-8.y.z-SNAPSHOT
15/05/21 09:39:49 INFO AbstractConnector: Started SocketConnector@0.0.0.0:44134
15/05/21 09:39:49 INFO Utils: Successfully started service 'HTTP file server' on port 44134.
15/05/21 09:39:49 INFO SparkEnv: Registering OutputCommitCoordinator
15/05/21 09:39:49 INFO Server: jetty-8.y.z-SNAPSHOT
15/05/21 09:39:49 WARN AbstractLifeCycle: FAILED SelectChannelConnector@0.0.0.0:4040: java.net.BindException: Adres jest już w użyciu
java.net.BindException: Adres jest już w użyciu
        at sun.nio.ch.Net.bind0(Native Method)
        at sun.nio.ch.Net.bind(Net.java:444)
        at sun.nio.ch.Net.bind(Net.java:436)
        at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:214)
        at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74)
        at org.spark-project.jetty.server.nio.SelectChannelConnector.open(SelectChannelConnector.java:187)
        at org.spark-project.jetty.server.AbstractConnector.doStart(AbstractConnector.java:316)
        at org.spark-project.jetty.server.nio.SelectChannelConnector.doStart(SelectChannelConnector.java:265)
        at org.spark-project.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:64)
        at org.spark-project.jetty.server.Server.doStart(Server.java:293)
        at org.spark-project.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:64)
        at org.apache.spark.ui.JettyUtils$.org$apache$spark$ui$JettyUtils$$connect$1(JettyUtils.scala:199)
        at org.apache.spark.ui.JettyUtils$$anonfun$2.apply(JettyUtils.scala:209)
        at org.apache.spark.ui.JettyUtils$$anonfun$2.apply(JettyUtils.scala:209)
        at org.apache.spark.util.Utils$$anonfun$startServiceOnPort$1.apply$mcVI$sp(Utils.scala:1837)
        at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:141)
        at org.apache.spark.util.Utils$.startServiceOnPort(Utils.scala:1828)
        at org.apache.spark.ui.JettyUtils$.startJettyServer(JettyUtils.scala:209)
        at org.apache.spark.ui.WebUI.bind(WebUI.scala:120)
        at org.apache.spark.SparkContext$$anonfun$12.apply(SparkContext.scala:309)
        at org.apache.spark.SparkContext$$anonfun$12.apply(SparkContext.scala:309)
        at scala.Option.foreach(Option.scala:236)
        at org.apache.spark.SparkContext.<init>(SparkContext.scala:309)
        at org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply$mcV$sp(ConsolidateSettingsSpec.scala:17)
        at org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply(ConsolidateSettingsSpec.scala:11)
        at org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply(ConsolidateSettingsSpec.scala:11)
        at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
        at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
        at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
        at org.scalatest.Transformer.apply(Transformer.scala:22)
        at org.scalatest.Transformer.apply(Transformer.scala:20)
        at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1647)
        at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
        at org.scalatest.FlatSpec.withFixture(FlatSpec.scala:1683)
        at org.scalatest.FlatSpecLike$class.invokeWithFixture$1(FlatSpecLike.scala:1644)
        at org.scalatest.FlatSpecLike$$anonfun$runTest$1.apply(FlatSpecLike.scala:1656)
        at org.scalatest.FlatSpecLike$$anonfun$runTest$1.apply(FlatSpecLike.scala:1656)
        at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
        at org.scalatest.FlatSpecLike$class.runTest(FlatSpecLike.scala:1656)
        at org.scalatest.FlatSpec.runTest(FlatSpec.scala:1683)
        at org.scalatest.FlatSpecLike$$anonfun$runTests$1.apply(FlatSpecLike.scala:1714)
        at org.scalatest.FlatSpecLike$$anonfun$runTests$1.apply(FlatSpecLike.scala:1714)
        at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
        at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
        at scala.collection.immutable.List.foreach(List.scala:318)
        at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
        at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:396)
        at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:483)
        at org.scalatest.FlatSpecLike$class.runTests(FlatSpecLike.scala:1714)
        at org.scalatest.FlatSpec.runTests(FlatSpec.scala:1683)
        at org.scalatest.Suite$class.run(Suite.scala:1424)
        at org.scalatest.FlatSpec.org$scalatest$FlatSpecLike$$super$run(FlatSpec.scala:1683)
        at org.scalatest.FlatSpecLike$$anonfun$run$1.apply(FlatSpecLike.scala:1760)
        at org.scalatest.FlatSpecLike$$anonfun$run$1.apply(FlatSpecLike.scala:1760)
        at org.scalatest.SuperEngine.runImpl(Engine.scala:545)
        at org.scalatest.FlatSpecLike$class.run(FlatSpecLike.scala:1760)
        at org.scalatest.FlatSpec.run(FlatSpec.scala:1683)
        at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:466)
        at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:677)
        at sbt.ForkMain$Run$2.call(ForkMain.java:294)
        at sbt.ForkMain$Run$2.call(ForkMain.java:284)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
15/05/21 09:39:49 WARN AbstractLifeCycle: FAILED org.spark-project.jetty.server.Server@ba32ce8: java.net.BindException: Adres jest już w użyciu
java.net.BindException: Adres jest już w użyciu
        at sun.nio.ch.Net.bind0(Native Method)
        at sun.nio.ch.Net.bind(Net.java:444)
        at sun.nio.ch.Net.bind(Net.java:436)
        at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:214)
        at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74)
        at org.spark-project.jetty.server.nio.SelectChannelConnector.open(SelectChannelConnector.java:187)
        at org.spark-project.jetty.server.AbstractConnector.doStart(AbstractConnector.java:316)
        at org.spark-project.jetty.server.nio.SelectChannelConnector.doStart(SelectChannelConnector.java:265)
        at org.spark-project.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:64)
        at org.spark-project.jetty.server.Server.doStart(Server.java:293)
        at org.spark-project.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:64)
        at org.apache.spark.ui.JettyUtils$.org$apache$spark$ui$JettyUtils$$connect$1(JettyUtils.scala:199)
        at org.apache.spark.ui.JettyUtils$$anonfun$2.apply(JettyUtils.scala:209)
        at org.apache.spark.ui.JettyUtils$$anonfun$2.apply(JettyUtils.scala:209)
        at org.apache.spark.util.Utils$$anonfun$startServiceOnPort$1.apply$mcVI$sp(Utils.scala:1837)
        at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:141)
        at org.apache.spark.util.Utils$.startServiceOnPort(Utils.scala:1828)
        at org.apache.spark.ui.JettyUtils$.startJettyServer(JettyUtils.scala:209)
        at org.apache.spark.ui.WebUI.bind(WebUI.scala:120)
        at org.apache.spark.SparkContext$$anonfun$12.apply(SparkContext.scala:309)
        at org.apache.spark.SparkContext$$anonfun$12.apply(SparkContext.scala:309)
        at scala.Option.foreach(Option.scala:236)
        at org.apache.spark.SparkContext.<init>(SparkContext.scala:309)
        at org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply$mcV$sp(ConsolidateSettingsSpec.scala:17)
        at org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply(ConsolidateSettingsSpec.scala:11)
        at org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply(ConsolidateSettingsSpec.scala:11)
        at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
        at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
        at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
        at org.scalatest.Transformer.apply(Transformer.scala:22)
        at org.scalatest.Transformer.apply(Transformer.scala:20)
        at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1647)
        at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
        at org.scalatest.FlatSpec.withFixture(FlatSpec.scala:1683)
        at org.scalatest.FlatSpecLike$class.invokeWithFixture$1(FlatSpecLike.scala:1644)
        at org.scalatest.FlatSpecLike$$anonfun$runTest$1.apply(FlatSpecLike.scala:1656)
        at org.scalatest.FlatSpecLike$$anonfun$runTest$1.apply(FlatSpecLike.scala:1656)
        at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
        at org.scalatest.FlatSpecLike$class.runTest(FlatSpecLike.scala:1656)
        at org.scalatest.FlatSpec.runTest(FlatSpec.scala:1683)
        at org.scalatest.FlatSpecLike$$anonfun$runTests$1.apply(FlatSpecLike.scala:1714)
        at org.scalatest.FlatSpecLike$$anonfun$runTests$1.apply(FlatSpecLike.scala:1714)
        at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
        at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
        at scala.collection.immutable.List.foreach(List.scala:318)
        at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
        at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:396)
        at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:483)
        at org.scalatest.FlatSpecLike$class.runTests(FlatSpecLike.scala:1714)
        at org.scalatest.FlatSpec.runTests(FlatSpec.scala:1683)
        at org.scalatest.Suite$class.run(Suite.scala:1424)
        at org.scalatest.FlatSpec.org$scalatest$FlatSpecLike$$super$run(FlatSpec.scala:1683)
        at org.scalatest.FlatSpecLike$$anonfun$run$1.apply(FlatSpecLike.scala:1760)
        at org.scalatest.FlatSpecLike$$anonfun$run$1.apply(FlatSpecLike.scala:1760)
        at org.scalatest.SuperEngine.runImpl(Engine.scala:545)
        at org.scalatest.FlatSpecLike$class.run(FlatSpecLike.scala:1760)
        at org.scalatest.FlatSpec.run(FlatSpec.scala:1683)
        at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:466)
        at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:677)
        at sbt.ForkMain$Run$2.call(ForkMain.java:294)
        at sbt.ForkMain$Run$2.call(ForkMain.java:284)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/stages/stage/kill,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/static,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/executors/threadDump/json,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/executors/threadDump,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/executors/json,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/executors,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/environment/json,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/environment,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/storage/rdd/json,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/storage/rdd,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/storage/json,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/storage,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/stages/pool/json,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/stages/pool,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/stages/stage/json,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/stages/stage,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/stages/json,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/stages,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/jobs/job/json,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/jobs/job,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/jobs/json,null}
15/05/21 09:39:49 INFO ContextHandler: stopped o.s.j.s.ServletContextHandler{/jobs,null}
15/05/21 09:39:49 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
15/05/21 09:39:49 INFO Server: jetty-8.y.z-SNAPSHOT
15/05/21 09:39:49 INFO AbstractConnector: Started SelectChannelConnector@0.0.0.0:4041
15/05/21 09:39:49 INFO Utils: Successfully started service 'SparkUI' on port 4041.
15/05/21 09:39:49 INFO SparkUI: Started SparkUI at http://localhost:4041
15/05/21 09:39:49 INFO Executor: Starting executor ID <driver> on host localhost
15/05/21 09:39:49 INFO AkkaUtils: Connecting to HeartbeatReceiver: akka.tcp://sparkDriver@localhost:51701/user/HeartbeatReceiver
15/05/21 09:39:49 INFO NettyBlockTransferService: Server created on 45485
15/05/21 09:39:49 INFO BlockManagerMaster: Trying to register BlockManager
15/05/21 09:39:49 INFO BlockManagerMasterActor: Registering block manager localhost:45485 with 2.8 GB RAM, BlockManagerId(<driver>, localhost, 45485)
15/05/21 09:39:49 INFO BlockManagerMaster: Registered BlockManager
15/05/21 09:39:49 WARN SparkContext: Multiple running SparkContexts detected in the same JVM!
org.apache.spark.SparkException: Only one SparkContext may be running in this JVM (see SPARK-2243). To ignore this error, set spark.driver.allowMultipleContexts = true. The currently running SparkContext was created at:
org.apache.spark.SparkContext.<init>(SparkContext.scala:80)
com.datastax.spark.connector.embedded.SparkTemplate$.resetSparkContext(SparkTemplate.scala:49)
com.datastax.spark.connector.embedded.SparkTemplate$.<init>(SparkTemplate.scala:60)
com.datastax.spark.connector.embedded.SparkTemplate$.<clinit>(SparkTemplate.scala)
org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply$mcV$sp(ConsolidateSettingsSpec.scala:13)
org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply(ConsolidateSettingsSpec.scala:11)
org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply(ConsolidateSettingsSpec.scala:11)
org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
org.scalatest.Transformer.apply(Transformer.scala:22)
org.scalatest.Transformer.apply(Transformer.scala:20)
org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1647)
org.scalatest.Suite$class.withFixture(Suite.scala:1122)
org.scalatest.FlatSpec.withFixture(FlatSpec.scala:1683)
org.scalatest.FlatSpecLike$class.invokeWithFixture$1(FlatSpecLike.scala:1644)
org.scalatest.FlatSpecLike$$anonfun$runTest$1.apply(FlatSpecLike.scala:1656)
org.scalatest.FlatSpecLike$$anonfun$runTest$1.apply(FlatSpecLike.scala:1656)
org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
org.scalatest.FlatSpecLike$class.runTest(FlatSpecLike.scala:1656)
        at org.apache.spark.SparkContext$$anonfun$assertNoOtherContextIsRunning$1$$anonfun$apply$10.apply(SparkContext.scala:1812)
        at org.apache.spark.SparkContext$$anonfun$assertNoOtherContextIsRunning$1$$anonfun$apply$10.apply(SparkContext.scala:1808)
        at scala.Option.foreach(Option.scala:236)
        at org.apache.spark.SparkContext$$anonfun$assertNoOtherContextIsRunning$1.apply(SparkContext.scala:1808)
        at org.apache.spark.SparkContext$$anonfun$assertNoOtherContextIsRunning$1.apply(SparkContext.scala:1795)
        at scala.Option.foreach(Option.scala:236)
        at org.apache.spark.SparkContext$.assertNoOtherContextIsRunning(SparkContext.scala:1795)
        at org.apache.spark.SparkContext$.setActiveContext(SparkContext.scala:1847)
        at org.apache.spark.SparkContext.<init>(SparkContext.scala:1754)
        at org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply$mcV$sp(ConsolidateSettingsSpec.scala:17)
        at org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply(ConsolidateSettingsSpec.scala:11)
        at org.apache.spark.sql.cassandra.ConsolidateSettingsSpec$$anonfun$1.apply(ConsolidateSettingsSpec.scala:11)
        at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
        at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
        at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
        at org.scalatest.Transformer.apply(Transformer.scala:22)
        at org.scalatest.Transformer.apply(Transformer.scala:20)
        at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1647)
        at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
        at org.scalatest.FlatSpec.withFixture(FlatSpec.scala:1683)
        at org.scalatest.FlatSpecLike$class.invokeWithFixture$1(FlatSpecLike.scala:1644)
        at org.scalatest.FlatSpecLike$$anonfun$runTest$1.apply(FlatSpecLike.scala:1656)
        at org.scalatest.FlatSpecLike$$anonfun$runTest$1.apply(FlatSpecLike.scala:1656)
        at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
        at org.scalatest.FlatSpecLike$class.runTest(FlatSpecLike.scala:1656)
        at org.scalatest.FlatSpec.runTest(FlatSpec.scala:1683)
        at org.scalatest.FlatSpecLike$$anonfun$runTests$1.apply(FlatSpecLike.scala:1714)
        at org.scalatest.FlatSpecLike$$anonfun$runTests$1.apply(FlatSpecLike.scala:1714)
        at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
        at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
        at scala.collection.immutable.List.foreach(List.scala:318)
        at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
        at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:396)
        at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:483)
        at org.scalatest.FlatSpecLike$class.runTests(FlatSpecLike.scala:1714)
        at org.scalatest.FlatSpec.runTests(FlatSpec.scala:1683)
        at org.scalatest.Suite$class.run(Suite.scala:1424)
        at org.scalatest.FlatSpec.org$scalatest$FlatSpecLike$$super$run(FlatSpec.scala:1683)
        at org.scalatest.FlatSpecLike$$anonfun$run$1.apply(FlatSpecLike.scala:1760)
        at org.scalatest.FlatSpecLike$$anonfun$run$1.apply(FlatSpecLike.scala:1760)
        at org.scalatest.SuperEngine.runImpl(Engine.scala:545)
        at org.scalatest.FlatSpecLike$class.run(FlatSpecLike.scala:1760)
        at org.scalatest.FlatSpec.run(FlatSpec.scala:1683)
        at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:466)
        at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:677)
        at sbt.ForkMain$Run$2.call(ForkMain.java:294)
        at sbt.ForkMain$Run$2.call(ForkMain.java:284)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)

**Comment by pkolaczk** [comment]
Additionally, the commit title had a typo in the ticket number. This would make it much harder to find the commit in the future.

**Pull request merged**